components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -1099,18 +1099,36 @@ fun stringOrNull(jsonObject: JSONObject, key: String): String? {
     }
 }
 
+enum class SearchResultReason(val reason: Int) {
```

Comment:
Instead of hardcoding the numbers, we can match on the protobuf type enum cases, like this:
https://github.com/mozilla/application-services/blob/47a84bc47e77f5b6b12144e5fe6374e540f8c4db/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/Device.kt#L29-L48

This is safer as we do not depend on magic numbers and the compiler will be able to catch future changes.

Reply:
I don't care much, but by far the most common change we've seen is adding a field, which this doesn't catch, no?

Reply:
Perhaps, use the not hard coded approach and throws an null pointer exception when any of the `reasons` falls under "`UNKNOWN`"?

Reply:
> I don't care much, but by far the most common change we've seen is adding a field, which this doesn't catch, no?

Depends on the language, Swift will catch non-exhaustive `switch` statements. I think Kotlin can also do that with some help.

Reply:
According to [Kotlin's guide](https://kotlinlang.org/docs/reference/control-flow.html),

> ...the else branch is mandatory, unless the compiler can prove that all possible cases are covered with branch conditions

I think removing the `else` branch will guarantee that new fields are covered while compiling. I tested by removing an existing branch, and the compiler did complain about a missing enum branch.